lib/screens/explore_screens/plant_seeds/interactor/mappers/user_balance_and_planted_state_mapper.dart

```diff
@@ -24,10 +25,10 @@ class UserBalanceAndPlantedStateMapper extends StateMapper {
       return currentState.copyWith(
         pageState: PageState.success,
         fiatAmount: rateState.fromSeedsToFiat(0, selectedFiat).fiatFormatted,
-        availableBalance: balance,
-        availableBalanceFiat: rateState.fromSeedsToFiat(balance?.quantity ?? 0, selectedFiat).fiatFormatted,
-        plantedBalance: plantedSeeds?.formattedQuantity,
-        plantedBalanceFiat: rateState.fromSeedsToFiat(plantedSeeds?.quantity ?? 0, selectedFiat).fiatFormatted,
+        availableBalance: TokenDataModel.from(balance?.quantity),
+        availableBalanceFiat: rateState.seedsToFiat(balance?.quantity ?? 0, selectedFiat),
+        plantedBalance: TokenDataModel.from(plantedSeeds?.quantity),
+        plantedBalanceFiat: rateState.seedsToFiat(plantedSeeds?.quantity ?? 0, selectedFiat),
```

Comment:
Planning for rateState and everything else dealing with currencies to have methods that take Fiat/Token data models as parameters, and return Fiat / Token data models.

Reply:
This will make handling additional tokens much easier also.